inspirehep/dojson/hep/fields/bd90x99x.py

```diff
@@ -35,11 +37,25 @@
 from ...utils import get_recid_from_ref, get_record_ref, strip_empty_values
 
 
+RE_VALID_PUBNOTE = re.compile(".*,.*,.*(,.*)?")
+
+
 @hep.over('references', '^999C5')
 def references(self, key, value):
     """Produce list of references."""
     value = utils.force_list(value)
 
+    def get_valid_pubnotes(pubnotes):
+        valid_pubnotes = []
+        raw_references = []
+        if pubnotes:
+            for pubnote in pubnotes:
+                if RE_VALID_PUBNOTE.match(pubnote):
+                    valid_pubnotes.append(pubnote)
+                else:
+                    raw_references.append(pubnote)
+        return valid_pubnotes, raw_references
```

Comment:
You could do this using `tee` from `itertools`: http://stackoverflow.com/a/949490/374865 (I'm not saying that you should, I just wanted to point out that this logic can be abstracted if it's needed in a lot of places).